code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,161 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	use_power = POWER_USE_OFF
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	var/obj/item/weapon/cell/cell
+	var/on = 0
+	var/drainage = 1
+	var/last_gurgle = 0
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+
+/obj/machinery/dehumidifier/New()
+	..()
+	cell = new/obj/item/weapon/cell/high(src)
+	update_icon()
+
+/obj/machinery/dehumidifier/on_update_icon(var/rebuild_overlay = 0)
+	if(!on)
+		icon_state = "dhum-off"
+	else
+		icon_state = "dhum-on"
+
+	if(rebuild_overlay)
+		overlays.Cut()
+		if(panel_open)
+			overlays  += "dhum-open"
+
+/obj/machinery/dehumidifier/examine(mob/user)
+	. = ..(user)
+
+	to_chat(user, "The dehumidifier is [on ? "on" : "off"] and the hatch is [panel_open ? "open" : "closed"].")
+	if(panel_open)
+		to_chat(user, "The power cell is [cell ? "installed" : "missing"].")
+	else
+		to_chat(user, "The charge meter reads [cell ? round(cell.percent(),1) : 0]%")
+	return
```

Comment:
Should try to avoid trailing returns, dunno if Loaf wants to enforce that though.